File: Corte_II/02-Retos/Retos-Busquedas/Reto-I-BusquedaRecursiva.py
#Busqueda Recursiva
#PREFERENCIALMENTE NO USARLA
#@Julian_H

def busqueda_recursiva(lista, objetivo, indice=0):
    if indice >= len(lista):
        return False
    elif lista[indice] == objetivo:
        return True
    else:
        return busqueda_recursiva(lista, objetivo, indice + 1)

lista_prueba = [1,2,3,4,5,6,7,8,9,10]
objetivo_prueba = 8
resultado_busqueda =  busqueda_recursiva(lista_prueba, objetivo_prueba, indice=0)
print(f"El resultado de la busqueda es: {resultado_busqueda}")

File: Corte_II/02-Retos/Retos-Busquedas/Reto-I-BusquedaRecursiva.cpp
//Busqueda recursiva
//@Julian_H
//PREFERIBLEMENTE NO USARLA
#include <iostream>

using namespace std;    

int busquedaRecursiva(int arr[], int n, int x) {
    if (n == 0) {
        return -1; // Elemento no encontrado
    }
    if (arr[n - 1] == x) {
        return n - 1; // Elemento encontrado en la posición n-1
    }
    return busquedaRecursiva(arr, n - 1, x); // Llamada recursiva con tamaño reducido
}

int main (){
    int arr[] = {2, 4, 6, 8, 10};
    int n = sizeof(arr) / sizeof(arr[0]);
    int x = 6;

    int resultado = busquedaRecursiva(arr, n, x);
    if (resultado != -1) {
        cout << "Elemento encontrado en la posicion: " << resultado << endl;
    } else {
        cout << "Elemento no encontrado" << endl;
    }

    return 0;
}